src/views/PositionPlot/PositionPlotView.tsx

```diff
@@ -52,35 +53,58 @@ const PositionPlotView: FunctionComponent<Props> = ({data, width, height}) => {
         context.stroke()
         context.setLineDash([]);
 
-        for (let dim of props.dimensions) {
-            context.strokeStyle = colorForUnitId(dim.dimensionIndex)
-            context.beginPath()
-            let first = true
-            for (let i=0; i<dim.pixelTimes.length; i++) {
-                const x = dim.pixelTimes[i]
-                const y = dim.pixelValues[i]
-                if (first) context.moveTo(x, y)
-                else context.lineTo(x, y)
-                first = false
-            }
-            context.stroke()
+        if (props.plotType === 'line' || !props.plotType) {
+            props.dimensions.forEach(dim => {
+                context.strokeStyle = colorForUnitId(dim.dimensionIndex)
+                context.lineWidth = 1.1 // hack--but fixes the 'disappearing lines' issue
+                context.beginPath()
+                dim.pixelTimes.forEach((x, ii) => {
+                    const y = dim.pixelValues[ii]
+                    ii === 0 ? context.moveTo(x, y) : context.lineTo(x, y)
+                })
+                context.stroke()
+            })
+        } else if (props.plotType === 'scatter') {
+            const circleWidthPx = 0.1 // could go even lighter
+            props.dimensions.forEach(d => {
+                context.fillStyle = colorForUnitId(d.dimensionIndex)
+                d.pixelTimes.forEach((t, ii) => {
+                    context.beginPath()
+                    context.arc(t, d.pixelValues[ii], circleWidthPx, 0, 2 * Math.PI)
+                    context.fill()
+                })
+            })
+        } else {
+            console.error(`Invalid plot type in PositionPlotView inputs: ${props.plotType}`)
         }
     }, [width])
 
     // Here we convert the native (time-based spike registry) data to pixel dimensions based on the per-panel allocated space.
-    const timeToPixelMatrix = use1dTimeToPixelMatrix(pixelsPerSecond, visibleTimeStartSeconds)
+    // const timeToPixelMatrix = use1dTimeToPixelMatrix(pixelsPerSecond, visibleTimeStartSeconds)
 
     const series = useMemo(() => {
         const series: {dimensionIndex: number, times: number[], values: number[]}[] = []
-        for (let dimensionIndex=0; dimensionIndex<data.dimensionLabels.length; dimensionIndex++) {
-            const filteredTimes = data.timestamps.filter(t => (visibleTimeStartSeconds <= t) && (t <= visibleTimeEndSeconds))
-            const filteredValues = data.timestamps.map((t, ii) => (data.positions[ii][dimensionIndex])).filter((a, ii) => (visibleTimeStartSeconds <= data.timestamps[ii]) && (data.timestamps[ii] <= visibleTimeEndSeconds))
+        const filteredTimeIndices = data.timestamps.flatMap((t, ii) => (visibleTimeStartSeconds <= t) && (t <= visibleTimeEndSeconds) ? ii : [])
+        const filteredTimes = filteredTimeIndices.map(i => data.timestamps[i])
+        const filteredValues = filteredTimeIndices.map(index => data.positions[index])
```

Comment:
Here I'm getting the filtered indices from the time series once, and then populating times and values from those indices. This avoids the need to filter on each loop & makes the logic a bit more straightforward vs. needing to filter one array based on looking up values in another.